thrall/test/lib/elasticsearch/ImageModelTest.scala

```diff
@@ -0,0 +1,64 @@
+package lib.elasticsearch
+
+import com.gu.mediaservice.lib.elasticsearch.MappingTest
+import com.gu.mediaservice.lib.logging.{LogMarker, MarkerMap}
+
+import scala.collection.TraversableLike
+import scala.concurrent.{Await, Future}
+
+class ImageModelTest extends ElasticSearchTestBase {
+  implicit val logMarker: LogMarker = MarkerMap()
+  "the image model matches the mapping" - {
+    "the testImage is fully populated (doesn't have any fields that are empty)" in {
+      /* This test is an important prelude to the serialisation check. Here we ensure that the
+       * test Image is fully populated with no None values or empty collections. This is to
+       * ensure that we test all corners of the data model when we serialise into elasticsearch.
+       * If this fails then it indicates that a field in the Image model does not contain a value.
+       * This should be fixed by adding appropriate values in MappingTest.scala
+       */
+
+      /* recurse through a typical scala structure */
+      def check(maybeProduct: Any): List[Either[String, Any]] = {
+        maybeProduct match {
+          case None =>
+            List(Left("Empty Option"))
+          case list: TraversableLike[_, _] if list.isEmpty =>
```

Comment:
Couldn't this be `list: Nil`?

Reply:
Nil is specifically an empty `List` so wouldn't match the other types of traversables. See https://github.com/scala/scala/blob/2.13.x/src/library/scala/collection/immutable/List.scala#L628-L640

Reply:
To be fair, the LHS shouldn't be named `list`. Will change to `collection` or something.

Reply:
I looked into this.... out of interest.  Nil will work for any *Sequence*, but not any *Traversable*.  This is obviously dumb, as an empty traversable (ie no fixed order) is clearly the same as an empty sequence (ie with fixed order).

For your delight:

```
scala> var x = Map()
x: scala.collection.immutable.Map[Nothing,Nothing] = Map()

scala> x.toSeq match { case Nil => println("empty"); case _ => println("other"); }
empty

scala> var x = None
x: None.type = None

scala> x.toSeq match { case Nil => println("empty"); case _ => println("other"); }
empty

scala> var x = Some("banana")
x: Some[String] = Some(banana)

scala> x.toSeq match { case Nil => println("empty"); case _ => println("other"); }
other
```